vyper/semantics/analysis/base.py

```diff
@@ -0,0 +1,267 @@
+import enum
+from dataclasses import dataclass
+from typing import Dict, List, Optional
+
+from vyper import ast as vy_ast
+from vyper.exceptions import (
+    CompilerPanic,
+    ImmutableViolation,
+    StateAccessViolation,
+    VyperInternalException,
+)
+from vyper.semantics.types.base import VyperType
+
+
+class _StringEnum(enum.Enum):
```

Comment:
A lot of code just to avoid writing the same name twice?

Reply:
that code was already there, just moved from vyper/semantics/types/utils.py: https://github.com/vyperlang/vyper/pull/2974/files#diff-2d8b16ee68553ace110d4070fab5d0f133beb1aef3a9acd1633951964600145bL41